src/aotcompile.cpp

```diff
@@ -735,6 +735,8 @@ void addOptimizationPasses(legacy::PassManagerBase *PM, int opt_level,
         PM->add(createLowerPTLSPass(dump_native));
         // Clean up write barrier and ptls lowering
         PM->add(createCFGSimplificationPass());
+        // Remove Julia's address space information
+        PM->add(createRemoveJuliaAddrspaces());
```

Comment:
Needs to be added here https://github.com/JuliaLang/julia/pull/35645/files#diff-474e444bd9ac3fcbffcfcecdbe3def83L621

Should We run InstCombine after?

Reply:
Added the pass there.

Which instruction combinations do you expect to get merged? It does already remove noop addrspacecasts. Can't hurt to try and move intrinsic lowering earlier in the PM to avoid other passes getting confused by address spaces.